src/features/staking/hooks/useApproveAst.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  useAccount,
+  useContractWrite,
+  usePrepareContractWrite,
+  useWaitForTransaction,
+} from "wagmi";
+import { ContractTypes } from "../../../config/ContractAddresses";
+import { useContractAddresses } from "../../../config/hooks/useContractAddress";
+import { astAbi } from "../../../contracts/astAbi";
+
+export const useApproveAst = ({
+  stakingAmount,
+  enabled = true,
+}: {
+  stakingAmount: number;
+  enabled?: boolean;
+}) => {
+  const { address } = useAccount();
+  const [airSwapToken] = useContractAddresses([ContractTypes.AirSwapToken], {
+    defaultChainId: 1,
+    useDefaultAsFallback: false,
+  });
+
+  const [airSwapStaking] = useContractAddresses(
+    [ContractTypes.AirSwapStaking],
+    {
+      defaultChainId: 1,
+      useDefaultAsFallback: false,
+    },
+  );
+
+  const { config: configApprove } = usePrepareContractWrite({
+    address: airSwapToken.address,
+    abi: astAbi,
+    functionName: "approve",
+    cacheTime: Infinity,
+    args: [airSwapStaking.address!, BigInt(+stakingAmount * Math.pow(10, 4))],
+    enabled: enabled && stakingAmount > 0 && !!address,
```

Comment:
Wherever you have used the non null assertion please also assert that it isn't null in the `enabled` property.

I.e. for this line add `&& !!airswapToken.address`

(P.s. this is not a result of using `!` instead of `as string`, the result is the same either way - the query can attempt to run without the address being set)

(P.p.s for this particular case it might be that the query can't run without address being set because the address in the function call args also happens to be the address of the contract and I think the hook does nothing if the contract address is undefined, but it's still good practice and there are instances in the PR that do need updating to prevent possible errors)